staging/src/k8s.io/client-go/rest/request.go

```diff
@@ -130,6 +130,7 @@ func NewRequest(client HTTPClient, verb string, baseURL *url.URL, versionedAPIPa
 		serializers: serializers,
 		backoffMgr:  backoff,
 		throttle:    throttle,
+		timeout:     timeout,
```

Comment:
Can you describe what you think this sets?  It actually sets a query parameter on our request for the apiserver to consume, not a timeout on the client side.  I doubt that's what most callers would expect.

Reply:
That's exactly my intention with this change. The downstream problem (https://bugzilla.redhat.com/show_bug.cgi?id=1433244) was that `--request-timeout` wasn't passed to apiserver, which although I've allowed this functionality to be long-running request, is being cut off by the apiserver.

Reply:
@deads2k the client-side timeout added by https://github.com/kubernetes/kubernetes/pull/33958 did not handles cases where the user requested a longer timeout, but the apiserver ended the request first. --request-timeout worked under the assumption that the client would be the one always terminating the connection

Reply:
> @deads2k the client-side timeout added by #33958 did not handles cases where the user requested a longer timeout, but the apiserver ended the request first. --request-timeout worked under the assumption that the client would be the one always terminating the connection

@juanvallejo more specifically.  You think using the same value for both is a good idea?

Reply:
From a user POV - I think the answer is yes. I'm not 100% sure about eventual downsides, though.

Reply:
I'm fine with having both match if the CLI behavior matches expectations.  @juanvallejo and @fabianofranz probably have more opinion there.

Is this set non-zero somewhere?  The code you updated all seems to set zero